### ISSUE TITLE
Library release support date overview

### ISSUE THREAD

**Issue by gavin-norman-sociomantic** [sociomantic-tsunami/neptune#29]
Other teams do not have an overview of when each library version is supported until. We discussed this in the leads meeting and people asked whether we could make some kind of easy overview.

(Moved from https://github.com/sociomantic/core/issues/25.)

**Comment by gavin-norman-sociomantic** [comment]
It just occurred to me that this could also replace the "maintained major branches" sections that we manually add to libraries' READMEs (e.g. [swarm](https://github.com/sociomantic-tsunami/swarm/#maintained-major-branches)). The initial release date of each major branch can be automatically parsed, as can the information about each library's support guarantees.

**Comment by leandro-lucarella-sociomantic** [comment]
Even when maybe is more cryptic to find out for users, I think this info could be added in a more parser friendly way to `.neptune.yml`. Maybe the README can point to that file to consult the dates? Maybe we could add a "badge" somehow that parses the file and shows this info embeded in the readme somehow?

**Comment by gavin-norman-sociomantic** [comment]
Yeah, any way to "generate" the support dates, instead of having to manually update them in the README each time a release is made would be amazing!

**Comment by mihails-strasuns-sociomantic** [comment]
@leandro-lucarella-sociomantic do we have a public docs.sociomantic.com counterpart?

**Comment by leandro-lucarella-sociomantic** [comment]
Nope, I wonder if we can use https://readthedocs.org/

**Comment by leandro-lucarella-sociomantic** [comment]
Oh, for the badges not really, I was thinking of general repos docs, sorry.

**Comment by leandro-lucarella-sociomantic** [comment]
Oh, for the badges not really, I was thinking of general repos docs, sorry.

**Comment by leandro-lucarella-sociomantic** [comment]
I'm a bit lost now. So, what is the exact feature request? To have a general overview of for how long versions will be maintained? If so I guess all we have to do is to define the support periods for each library and then the new neptune tool could calculate the support windows in terms of dates based on that, right? So we can have a more cryptic and hidden config (that is not that useful for the end user) saying something like:

```yml
support-guarantees:
    major-months: 6
    minor-versions: 2
```

So based on that the neptune tool could calculate the support windows and put them in each release notes maybe? Would that be enough?

**Comment by leandro-lucarella-sociomantic** [comment]
Maybe even saying in each release notes how much time is left, to make it more clear, like "Supported until XX/XX/XXXX (2 months, 3 days left)"

**Comment by gavin-norman-sociomantic** [comment]
You mean that this information in the release notes would be generated each time someone looks at the release, or something?

**Comment by mihails-strasuns-sociomantic** [comment]
I think ideal solution would be a single badge/iframe per repo which shows all versions with their support dates (and possibly transitive dependencies for extra wow) which gets linked from README so that it always stays up to date in all branches without any manual maintenance.

**Comment by leandro-lucarella-sociomantic** [comment]
> You mean that this information in the release notes would be generated each time someone looks at the release, or something?

No, I was saying to make it static as part of the release notes at the time the release happens.

**Comment by leandro-lucarella-sociomantic** [comment]
Seems like the simplest solution to me, but if you wanted a single place with all the supported versions and the supported period for each, then that would not do it.

**Comment by mathias-baumann-sociomantic** [comment]
In regards to https://github.com/sociomantic-tsunami/neptune/issues/70, I would propose to add

```
maintained_minor_count
maintained_major_months
```

 to neptune.yml.

First being how many minor releases are supported and 
second being how many months after a major release the previous major will be supported.

**Comment by leandro-lucarella-sociomantic** [comment]
Why not https://github.com/sociomantic-tsunami/neptune/issues/29#issuecomment-305175217? I prefer to have a dedicated "section" and not repeating `maintained` (and `-` instead of `_`). I don't mind the `count` vs `versions`.
Since we can put them in a "section", why repeating the `maintained`? Also

**Comment by mathias-baumann-sociomantic** [comment]
My version seems easier to parse. But I don't really care, fine with either.

**Comment by mihails-strasuns-sociomantic** [comment]
You are not going to parse yaml manually, aren't you? ;)
Just in case - neptune already depends on D-YAML and it is used from neptune-overview.

**Comment by mathias-baumann-sociomantic** [comment]
At the beginning I was just doing string-match ;). The simplest tool that gets the job done.
Later I needed to filtert by type=library and at that point I switched to use D-YAML.
